pennylane/templates/embeddings.py

```diff
@@ -0,0 +1,230 @@
+    """
+
+    if not isinstance(wires, Iterable):
+        raise ValueError("Wires needs to be a list of wires that the embedding uses, got {}.".format(wires))
+
+    if len(features) > len(wires):
+        raise ValueError("Number of features to embed cannot be larger than number of wires which is {}, "
+                         "got {}.".format(len(wires), len(features)))
+    if rotation == 'X':
+        for f, w in zip(features, wires):
+            RX(f, wires=w)
+
+    elif rotation == 'Y':
+        for f, w in zip(features, wires):
+            RY(f, wires=w)
+
+    elif rotation == 'Z':
+        for f, w in zip(features, wires):
+            RZ(f, wires=w)
+
+    else:
+        raise ValueError("Rotation has to be `X`, `Y` or `Z`, got {}.".format(rotation))
+
+
+def AmplitudeEmbedding(features, wires=None):
+    """
+    Encodes :math:`2^n` features into the amplitude vector of :math:`n` qubits.
+
+    The absolute square of all  elements in ``features`` has to add up to one.
+
+    .. note::
+
+        AmplitudeEmbedding uses PennyLane's :class:`~.QubitStateVector` and only works in conjunction with
+        devices that implement this function.
+
+    Args:
+        features (array): Input array of shape ``(2**n, )``
+
+    Keyword Args:
+        wires (Sequence[int]): sequence of qubit indices that the template acts on
+    """
+
+    if not isinstance(wires, Iterable):
+        raise ValueError("Wires needs to be a list of wires that the embedding uses, got {}.".format(wires))
+
+    if 2**len(wires) != len(features):
+        raise ValueError("AmplitudeEmbedding requires a feature vector of size 2**n_qubits which is {}, "
+                         "got {}.".format(2 ** len(wires), len(features)))
+
+    QubitStateVector(features, wires=wires)
+
+
+def BasisEmbedding(features, wires=None):
+    r"""Encodes :math:`n` binary features into a basis state of :math:`n` qubits.
+
+    For example, for ``features=[0, 1, 0]``, the quantum system will be prepared in state :math:`|010 \rangle`.
+
+    .. note::
+
+        BasisEmbedding uses PennyLane's :class:`~.BasisState` and only works in conjunction with
+        devices that implement this function.
+
+    Args:
+        features (array): Binary input array of shape ``(n, )``
+
+    Keyword Args:
+        wires (Sequence[int]): sequence of qubit indices that the template acts on
+    """
+    if not isinstance(wires, Iterable):
+        raise ValueError("Wires needs to be a list of wires that the embedding uses, got {}.".format(wires))
+
+    if len(features) > len(wires):
+        raise ValueError("Number of bits to embed cannot be larger than number of wires which is {}, "
+                         "got {}.".format(len(wires), len(features)))
+    BasisState(features, wires=wires)
+
+
+def SqueezingEmbedding(features, execution='amplitude', c=0.1, wires=None):
+    r"""Encodes :math:`M` features into the squeezing amplitudes :math:`r \geq 0` or phases :math:`\phi \in [0, 2\pi)`
```

Comment:
Changed SqueezingEmbedding and DisplacementEmbedding to work like AngleEmbedding, which means that it can take fewer features than wires. Updated docs and errors + tests.